runner/src/run.rs

```diff
@@ -180,3 +200,49 @@ fn launch_spike(args: &RunArgs, miralis: PathBuf, firmware: PathBuf) -> ExitCode
 fn raw_to_elf(raw_path: &str) -> &str {
     &raw_path[..raw_path.len() - 4]
 }
+
+fn fetch_ubuntu_image() {
+    let file_path = "misc/ubuntu-miralis.img";
+
+    if !Path::new(file_path).exists() {
+        // Download image
+        log::info!("Ubuntu image not found. Fetching the image...");
+        let mut image_downloader = Command::new("wget");
+        image_downloader.arg("https://cdimage.ubuntu.com/releases/24.04/release/ubuntu-24.04.1-preinstalled-server-riscv64.img.xz");
+        image_downloader
+            .status()
+            .expect("Failed to downlaod Ubuntu");
+
+        // Extract image
+        log::info!("Extracting the image");
+        let mut image_extractor = Command::new("xz");
+        image_extractor
+            .arg("-dk")
+            .arg("ubuntu-24.04.1-preinstalled-server-riscv64.img.xz");
+        image_extractor.status().expect("Failed to downlaod Ubuntu");
+
+        // Move and clean
+        log::info!("Moving and cleaning the download");
+        let mut mover = Command::new("mv");
+        mover
+            .arg("ubuntu-24.04.1-preinstalled-server-riscv64.img")
+            .arg("misc/ubuntu-miralis.img");
+        mover.status().expect("Failed to downlaod Ubuntu");
+
+        let mut remove_extracted_download = Command::new("rm");
+        remove_extracted_download.arg("ubuntu-24.04.1-preinstalled-server-riscv64.img");
+        remove_extracted_download
+            .status()
+            .expect("Failed to remove extracted download");
+
+        let mut remove_raw_download = Command::new("rm");
+        remove_raw_download.arg("ubuntu-24.04.1-preinstalled-server-riscv64.img.xz");
+        remove_raw_download
+            .status()
+            .expect("Failed to remove raw download");
+
+        log::info!("Image ready to use");
+    } else {
+        log::debug!("Ubuntu image already exists.");
+    }
+}
```

Comment:
This is something we want to move in `artifact.rs`, as it deals with downloading stuff.

In its current form that is very specific to Ubuntu, which would create a lot of tech debt as soon as we want to include other disk images. For instance we will probably want to support [Alpine Linux](https://alpinelinux.org/downloads/) for testing (because it is much lighter than Ubuntu).

What I would suggest is to extend the artifact manifest to add a new kind of artifacts (`disk`) and then allow specifying a disk under the `[qemu]` table in the config file. Then we can add some code in the Runner's `artifact.rs` to handle the download from a given URL specified in the artifact manifest.